realm/realm-library/src/main/java/io/realm/RealmInteger.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2017 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm;
+
+import io.realm.annotations.Beta;
+
+
+/**
+ * TODO: document the behaviour of this object, as soon as the implementation defines it.
+ */
+@Beta
+public final class RealmInteger extends Number implements Comparable<RealmInteger> {
+    private long value;
+
+    /**
+     * Creates a new {@code RealmInteger}, with the specified initial value.
+     *
+     * @param value initial value.
+     */
+    public RealmInteger(long value) {
+        this.value = value;
+    }
+
+    /**
+     * Creates a new realm integer, with the specified initial value.
+     *
+     * @param value initial value: parsed by {@code Long.parseLong}.
+     */
+    public RealmInteger(String value) {
+        this.value = Long.parseLong(value);
+    }
+
```

Comment:
Not quite sure what you mean.  Java fields are initialized to 0, so `this.value` is 0, until it is set.  That shouldn't matter, though, because `Long.parseLong` throws an (unchecked) exception if it cannot parse its argument.   If you call `new RealmInteger("can't parse me!")` the object will never be constructed, so default values shouldn't be necessary...